src/amo/components/SearchResult/index.js

```diff
@@ -0,0 +1,101 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { compose } from 'redux';
+
+import fallbackIcon from 'amo/img/icons/default-64.png';
+import Link from 'amo/components/Link';
+import translate from 'core/i18n/translate';
+import { ADDON_TYPE_THEME } from 'core/constants';
+import { isAllowedOrigin, sanitizeHTML } from 'core/utils';
+import Icon from 'ui/components/Icon';
+import Rating from 'ui/components/Rating';
+
+import './styles.scss';
+
+
+export class SearchResultBase extends React.Component {
+  static propTypes = {
+    addon: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+  }
+
+  render() {
+    const { addon, i18n } = this.props;
+    const averageDailyUsers = addon.average_daily_users;
+    const isTheme = addon.type === ADDON_TYPE_THEME;
+    const resultClassnames = classNames('SearchResult', {
+      'SearchResult--theme': isTheme,
+    });
+
+    // Fall-back to default icon if invalid icon url.
+    const iconURL = isAllowedOrigin(addon.icon_url) ? addon.icon_url : fallbackIcon;
+    const themeURL = (addon.theme_data && isAllowedOrigin(addon.theme_data.previewURL)) ?
+      addon.theme_data.previewURL : null;
+    const imageURL = isTheme ? themeURL : iconURL;
+
+    // Sets classes to handle fallback if theme preview is not available.
+    const iconWrapperClassnames = classNames('SearchResult-icon-wrapper', {
+      'SearchResult-icon-wrapper--no-theme-image': (
+        isTheme && imageURL === null
+      ),
+    });
+
+    // eslint-disable react-no-danger
+    return (
+      <li className={resultClassnames}>
+        <Link to={`/addon/${addon.slug}/`}
+              className="SearchResult-link"
+              ref={(el) => { this.name = el; }}>
+          <div className={iconWrapperClassnames}>
+            {imageURL ? (
+              <img className="SearchResult-icon" src={imageURL} alt="" />
+            ) : (
+              <p className="SearchResult-notheme">
+                {i18n.gettext('No theme preview available')}
+              </p>
+            )}
+          </div>
+
+          <div className="SearchResult-contents">
+            <h2 className="SearchResult-name">{addon.name}</h2>
+            <p
+              className="SearchResult-summary"
+              dangerouslySetInnerHTML={sanitizeHTML(addon.summary)}
```

Comment:
Though I opted not to use description here because it has the potential to be very long and I don't want all that text on the search results page. We will deal with it better in #2626 where if we find a description but no summary we can make a shortened summary from description I think. But for now they are fine blank.